docs/design/proposals/server-side-override/server-side-override.md

```diff
@@ -0,0 +1,224 @@
+the details of server-side-overrides in the custom resource’s Annotation, add a condition in custom resource  status to 
+indicate that server override some spec fieds.
+
+> Note: The original value provided by the k8s user in desired spec will not be updated, even though server udated that value. 
+
+
+1. *Consistency with K8s conventions* ✅
+    1. In K8s convention, the spec fields are not duplicated into status fields.
+2. *Consistency across ACK service controllers* ✅
+3. *GitOps compatibility* ✅
+    1. There should be no complexity in implementing GitOps for ACK custom resources. 
+    2. Approach mentioned in low level design below should work.
+4. *User experience for custom resource lifecycle* ✅
+    1. Applying updates to custom resource will not be confusing for the user. Users will update the spec and service
+     controller will create Update request based on new spec.
+
+### High Level Design
+
+![High Level Design](./images/server-side-override.png)
+[source](./images/server-side-override.drawio)
+
+
+### Low Level Design
+
+To implement the flow described in high level design, we will walk through reconciliation of an example resource through
+following steps and layout in detail how these steps will be achieved by ACK controller. 
+We assume that Fluxcd is used to apply desired state of custom resource.
+
+1. Assume there is a custom resource with following spec and status fields.
+
+    “fieldA” and “fieldC” are regular spec and status field respectively. “fieldA” is provided by K8s user and “fieldD” 
+    is updated in status based on latest observation.
+    
+    overrideFieldB is modified by reconciler whenever the value is provided by the k8s user. Ex: “-xyz” gets appended to
+    it’s value. 
+
+    ```yaml
+    Spec:
+     fieldA string
+     overrideFieldB string
+    Status:
+     fieldC string
+    ```
+
+    and the manifest file(desired state) looks like,
+
+    ```yaml
+    Spec:
+     fieldA: valueA
+     overrideFieldB: valueB
+   ```
+
+2.  When this resource is reconciled by the ACK service controller, we expect the CreateOutput to have following value 
+    for these fields. fieldA → “valueA“, overrideFieldB → “valueB-xyz“, fieldC → ”valueC“
+    
+    Based on this result, ACK service controller will persist the following object in etcd:
+    
+    Annotations:
+    
+    ```yaml
+    services.k8s.aws/spec-overrides: 
+       “{ 
+           \“overrideFieldB\”:
+               {
+                   \“userInput\”:\“valueB\”,
+                   \“serverOverride\”:\“valueB-xyz\”
+               }
+       }”
+       Spec:
+        fieldA: valueA
+        overrideFieldB: valueB
+       Status:
+        fieldC: valueC
+        conditions:
+        - type: ACK.ServerSideOverride
+          status: True
+          reason: ServerSideSpecOverride
+          message: Spec override values are present in services.aws.k8s/spec-overrides annotation
```

Comment:
Could this message include a list of overridden keys, saving users the hassle of looking up the annotations.